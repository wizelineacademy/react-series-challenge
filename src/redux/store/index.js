import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../reducers';
import customMiddleware from '../middleware';
import createSagaMiddleware from 'redux-saga';
import rootSaga from '../sagas';

const sagaMiddleware = createSagaMiddleware();

const initialState = {};
const enhancers = [];
const middleware = [
  sagaMiddleware,
  ...customMiddleware,
];

const composedEnhancers = compose(
  applyMiddleware(...middleware),
  ...enhancers,
);

const store = createStore(
  rootReducer,
  initialState,
  composedEnhancers,
);

sagaMiddleware.run(rootSaga);

export default store;
