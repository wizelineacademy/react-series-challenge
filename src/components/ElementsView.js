import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {loadAllGifs} from '../actions/allGifs';
import ElementList from './ElementList';

class ElementsView extends Component{

    componentDidMount() {
        this.props.loadAllGifs();
    }

    render(){
        return (
            <div>
                <h4>Main Gifs View!</h4>
                <ElementList all={this.props.allGifs} search={this.props.searchGifs} />
            </div>
        );
    }

}

//Set the main stage to props i need to use on this component
const mapStateToProps = (state) => {
    const {
        allGifs,
        searchGifs
    } = state;

    return {
        allGifs,
        searchGifs
    };
};

//Set the acction is going to be trigerred on this component
const mapDispatchToProps = (dispatch) => {

    return bindActionCreators({
        loadAllGifs,
    }, dispatch);

};

export default connect(mapStateToProps, mapDispatchToProps)(ElementsView);