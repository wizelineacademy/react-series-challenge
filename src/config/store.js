import {createStore, applyMiddleware, compose} from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas';
import {APPLICATION_LOCAL_STORAGE_KEY} from './config';
import {initialState as gifInitialState} from '../reducers/gif';

// Middlewares
const sagaMiddleware = createSagaMiddleware();
const middlewares = [
  sagaMiddleware
];

// Enhancers
const createEnhancers = () => {
  let enhancers = [];

  /**
   * Uncomment this lines and add the redux-devtools package to enable redux dev
   *  tools into development mode.
   */
  // if (process.env.NODE_ENV !== 'production') {
  //   enhancers.push(
  //     window.__REDUX_DEVTOOLS_EXTENSION__
  //       && window.__REDUX_DEVTOOLS_EXTENSION__());
  // }

  return enhancers;
};

const persistStateDetails = (state) => {
  try {
    let stateCopy = JSON.parse(JSON.stringify(state));

    // Remove unnecesary state details
    stateCopy.gifs.fetching = gifInitialState.fetching;
    stateCopy.gifs.displayed = gifInitialState.displayed;
    stateCopy.gifs.favoritesTextFilter = gifInitialState.favoritesTextFilter;

    let serializedState = JSON.stringify(stateCopy);
    localStorage.setItem(APPLICATION_LOCAL_STORAGE_KEY, serializedState);
  } catch (e) {}
}

const retrievePreviousStateDetails = () => {
  let result = {};

    try {
      let serializedState =
        window.localStorage.getItem(APPLICATION_LOCAL_STORAGE_KEY);

      if (serializedState) {
        result = JSON.parse(serializedState);
      }
    } catch (e) {}

  return result;
}

const enhancers = createEnhancers();
const composedEnhancers = (enhancers.length > 0)
  ? compose(applyMiddleware(...middlewares), ...enhancers)
  : applyMiddleware(...middlewares);


const initializeStore = (initialState) => {
  if (!initialState) {
    // Rehydrate the store
    initialState = retrievePreviousStateDetails();
  }

  // Store object
  const store = createStore(rootReducer, initialState, composedEnhancers);

  store.subscribe(() => persistStateDetails(store.getState()));

  sagaMiddleware.run(rootSaga);

  return store;
}

export default initializeStore;
