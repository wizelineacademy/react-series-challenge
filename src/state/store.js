import createSagaMiddleware from "redux-saga";
import { createStore, applyMiddleware } from "redux";

import rootSaga from "./sagas";
import reducers from "./reducers";

const sagaMiddleware = createSagaMiddleware();

const store = createStore(reducers, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga);

export default store;
