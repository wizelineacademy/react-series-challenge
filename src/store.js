import { createStore, applyMiddleware, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import { gif, trending } from './reducers';

const reducer = combineReducers({ gif, trending });

export default createStore(reducer, composeWithDevTools(applyMiddleware(thunk)));
