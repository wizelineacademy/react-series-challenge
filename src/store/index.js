import { createStore, applyMiddleware, compose } from 'redux';
import combineReducers from '../reducers';
import createSagaMiddleware from 'redux-saga';
import rootSaga from '../sagas';
import customMiddleware from '../middleware';
import logger from 'redux-logger'


const initialState   = {};
const enhancers = [];
const sagaMiddleware = createSagaMiddleware();

const middleware = [
    sagaMiddleware,
    logger,
    ...customMiddleware
];

const composedEnhancers = compose(
    applyMiddleware(...middleware),
    ...enhancers
);

const index = createStore(
    combineReducers,
    initialState,
    composedEnhancers
);
sagaMiddleware.run(rootSaga);
export default index;