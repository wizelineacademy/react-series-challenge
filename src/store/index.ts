import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga'
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from '../reducers';
import gifsFetcher from '../middleware';
import rootSaga from '../sagas/';
import autosave from '../middleware/autosave';

const sagaMiddleware = createSagaMiddleware();

function _getInitialFavs() {
  const data = localStorage.getItem('favs') || "";
  try {
    return JSON.parse(data);
  } catch(_) {
    return {}
  }
}

const initialState = {
  favs: _getInitialFavs()
};
const middleware = [
  sagaMiddleware,
  autosave,
  gifsFetcher,
];

const composedEnhancers = compose(
  applyMiddleware(...middleware),
);

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(composedEnhancers),
);

sagaMiddleware.run(rootSaga);

export default store;
