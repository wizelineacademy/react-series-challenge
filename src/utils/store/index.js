import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../redux/reducer';
import rootSaga from '../redux/saga';
import customMiddleware from '../middleware';

const initialState = {};
const enhancers = [
];
const middleware = [
  ...Object.values(customMiddleware),
];

const composedEnhancers = compose(
  applyMiddleware(...middleware),
  window.devToolsExtension ? window.devToolsExtension() : f => f,
  ...enhancers,
);

const store = createStore(
  rootReducer,
  initialState,
  composedEnhancers,
);

customMiddleware.sagaMiddleware.run(rootSaga);

export default store;
