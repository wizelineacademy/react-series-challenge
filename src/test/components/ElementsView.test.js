import React from 'react';
import { Provider } from 'react-redux';
import ElementsView from '../../components/ElementsView';
import renderer from 'react-test-renderer';
import store from '../../store/index';
import localStorageMock from './localStorageMock';

describe('ElementsView', () => {

    it('should render without problems', () => {
        const wrapper = renderer.create(
            <Provider store={store}>
                <ElementsView />
            </Provider>
        ).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

});
