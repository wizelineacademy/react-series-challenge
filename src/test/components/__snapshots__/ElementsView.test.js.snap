// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementsView should render without problems 1`] = `
<div>
  <h2>
    Type to search for gifs
  </h2>
  <h4>
    Main Gifs View!
  </h4>
  <div
    className="sc-EHOje gqTaLh"
  />
</div>
`;
